### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by Haywayaheadshot**
In this milestone I:

- Created app.rb and main.rb files with their respective classes.
- Added an interface class to the main.rb file.
- Created a cases method to selected the users selection.
- Created methods to:
- List all books.
- List all people.
- Create a person (teacher or student, not a plain Person).
- Create a book.
- Create a rental.
- List all rentals for a given person id.
- Fixed and refactored the code to pass linters.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    puts 'Select an option: '
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals'
+    puts '7 - Exit'
+    puts '--------------------------------------------------'
+  end
```
Comment (jerryowusu) [inline app.rb:19]: Your app is currently not storing the data anywhere so it closes when you create a person or book. As a result it becomes difficult to go ahead and create  rental or even list available books/people. 

- [ ] Kindly consider initializing the books, people, and rentals and storing them in an array to fix this issue. 
- [ ] You should also check and make sure to the app can list all books and able to undergo all functions successfully. :pray: 

Screenshot from 2023-01-06 23-14-12

Screenshot from 2023-01-06 23-14-19

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  def list_all_people
+    puts 'There\'s no people added yet' if Student.all.empty? && Teacher.all.empty?
+    Student.all.each do |student|
+      puts %([Student] Name: #{student.name}, ID: #{student.id}, Age: #{student.age})
+    end
+    Teacher.all.each do |teacher|
+      puts %([Teacher] Name: #{teacher.name}, ID: #{teacher.id}, Age: #{teacher.age})
+    end
+  end
```
Comment (jerryowusu) [inline app.rb:36]: The app abruptly closes when there is no book or people availabe. This is not supposed to be so, it should allow you to create a book or person and also check again to create rentals. Kindly check and fix this issue. 

Screenshot from 2023-01-06 23-23-31


Screenshot from 2023-01-06 23-23-22

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @Haywayaheadshot , 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

so close


## Highlights
All linters are passing :heavy_check_mark: 
Great effort so far :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Haywayaheadshot** [comment]
@jerryowusu Thank you for reviewing my project. I will work on the changes asap

**Review by mwanawabangona — APPROVED** [review]
Hi @Haywayaheadshot 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Haywayaheadshot** [comment]
@mwanawabangona Thank you for reviewing my project. Have a great weekend

**Pull request merged**